src/ecies.cpp

```diff
@@ -299,6 +299,12 @@ class ECIESEncryptionCtx::Impl : public ECIESImpl
 
 };
 
+size_t getEcPublicKeyOctetLength(const AsymmetricPrivateKey &privKey, openssl::EllipticCurvePointConversionForm form) {
```

Comment:
This method should check if this is actually an EC key and throw otherwise in order to prevent accidental misuse.